_posts/2020-02-20-ddl.md

```diff
@@ -0,0 +1,124 @@
+
+<div class="alert alert-info">
+If your favorite system does not have a connector maintained in the central Apache Flink repository, check out the <a href="https://flink-packages.org">flink packages website</a>, which has a growing number of community-maintained components.
+</div>
+
+While these connectors are battle-tested and production-ready, they are written in Java and configured in code, which means they are not amenable to pure SQL or Table applications.
+Relational queries should execute against relational tables.
+
+# CREATE TABLE Statements
+
+While Flink SQL has long provided table abstractions atop some of Flink's most popular connectors, configurations were not always so straightforward.
+Begining in 1.10, Flink supports defining tables through `CREATE TABLE` statements.
+With this feature, users can now create logical tables, backed by various external systems, in pure SQL. 
+
+This means you can define a logical table that can be read from or written to that is abstracted away from its physical data store.
+Coupled with Flink SQL's unified approach to batch and stream processing, Flink provides a straight line from discovery to production.
+
+Users can define tables over static data sets, anything from a local CSV file to a full-fledged data lake or even Hive.
+Leveraging Flink's efficient batch processing capabilities, they can perform ad-hoc queries searching for exciting insights.
+Once something interesting is identified, businesses can gain real-time and continuous insights by merely altering the table so that it is powered by a message queue such as Kafka.
+Because Flink guarantees SQL queries have unified semantics over batch and streaming, users can be confident that redeploying this query as a continuous streaming application over a message queue will output identical results.
+
+{% highlight sql %}
+-- Define a table called orders that is backed by a Kafka topic
+-- The definition includes all relevant Kafka properties,
+-- the underlying format (JSON) and even defines a
+-- watermarking algorithm based on one of the fields
+-- so that this table can be used with event time.
+CREATE TABLE orders (
+	user_id    BIGINT,
+	product    STRING,
+	order_time TIMESTAMP(3),
+	WATERMARK FOR order_time AS order_time - '5' SECONDS
+) WITH (
+	'connector.type'    	 = 'kafka',
+	'connector.version' 	 = 'universal',
+	'connector.topic'   	 = 'orders',
+	'connector.startup-mode' = 'earliest-offset',
+	'connector.properties.bootstrap.servers' = 'localhost:9092',
+	'format.type' = 'json' 
+);
+
+-- Define a table called product_analsysis
+-- on top of ElasticSearch 7 where  we 
+-- can write the results of our query. 
+CREATE TABLE product_analysis (
+	product 	STRING,
+	tracking_time 	TIMESTAMP(3),
+	units_sold 	BIGINT
+) WITH (
+	'connector.type'    = 'elasticsearch',
+	'connector.version' = '7',
+	'connector.hosts'   = 'http://host_name:9092',
```

Comment:
Change to ES default port?

Reply:
I took this from the flink docs, I will follow up with an update there